Tests/Services/HealthTests/HealthTests.swift

```diff
@@ -0,0 +1,322 @@
+ * limitations under the License.
+ */
+
+import GRPCHealth
+import GRPCInProcessTransport
+import XCTest
+
+@available(macOS 15.0, iOS 18.0, watchOS 11.0, tvOS 18.0, visionOS 2.0, *)
+final class HealthTests: XCTestCase {
+  private func withHealthClient(
+    _ body: @Sendable (Grpc_Health_V1_HealthClient, Health.Provider) async throws -> Void
+  ) async throws {
+    let health = Health()
+    let inProcess = InProcessTransport.makePair()
+    let server = GRPCServer(transport: inProcess.server, services: [health.service])
+    let client = GRPCClient(transport: inProcess.client)
+    let healthClient = Grpc_Health_V1_HealthClient(client: client)
+
+    try await withThrowingDiscardingTaskGroup { group in
+      group.addTask {
+        try await server.run()
+      }
+
+      group.addTask {
+        try await client.run()
+      }
+
+      do {
+        try await body(healthClient, health.provider)
+      } catch {
+        XCTFail("Unexpected error: \(error)")
+      }
+
+      group.cancelAll()
+    }
+  }
+
+  func testCheckOnKnownService() async throws {
+    try await withHealthClient { (healthClient, healthProvider) in
+      let testServiceDescriptor = ServiceDescriptor.testService
+
+      healthProvider.updateStatus(.serving, ofService: testServiceDescriptor)
+
+      var message = Grpc_Health_V1_HealthCheckRequest()
+      message.service = testServiceDescriptor.fullyQualifiedService
+
+      try await healthClient.check(request: ClientRequest.Single(message: message)) { response in
+        try XCTAssertEqual(response.message.status, .serving)
+      }
+    }
+  }
+
+  func testCheckOnUnknownService() async throws {
+    try await withHealthClient { (healthClient, healthProvider) in
+      var message = Grpc_Health_V1_HealthCheckRequest()
+      message.service =
+        ServiceDescriptor(package: "does.not", service: "Exist")
+        .fullyQualifiedService
+
+      try await healthClient.check(request: ClientRequest.Single(message: message)) { response in
+        try XCTAssertThrowsError(ofType: RPCError.self, response.message) { error in
+          XCTAssertEqual(error.code, .notFound)
+        }
+      }
+    }
+  }
+
+  func testCheckOnServer() async throws {
+    try await withHealthClient { (healthClient, healthProvider) in
+      healthProvider.updateStatus(.notServing, ofService: ServiceDescriptor.server)
+
+      let message = Grpc_Health_V1_HealthCheckRequest()
+
+      try await healthClient.check(request: ClientRequest.Single(message: message)) { response in
+        try XCTAssertEqual(response.message.status, .notServing)
+      }
+    }
+  }
+
+  func testWatchOnKnownService() async throws {
+    try await withHealthClient { (healthClient, healthProvider) in
+      let testServiceDescriptor = ServiceDescriptor.testService
+
+      let statusesToBeSent: [ServingStatus] = [.serving, .notServing, .serving]
+
+      healthProvider.updateStatus(statusesToBeSent[0], ofService: testServiceDescriptor)
```

Comment:
Can you add a comment as to why we send this status before starting the RPC because that's quite important.